Apps/WebClient/src/ClientApp/src/services/restDependentService.ts

```diff
@@ -33,21 +33,25 @@ export class RestDependentService implements IDependentService {
         hdid: string,
         dependent: AddDependentRequest
     ): Promise<AddDependentRequest> {
-        return new Promise((resolve, reject) => {
+        return new Promise<AddDependentRequest>((resolve, reject) => {
             if (!this.isEnabled) {
-                resolve();
+                reject("dependent module is disabled.");
```

Comment:
This will show errors when the module is disabled. Instead, it needs to return an empty object sucessfully. Same on other locations

Reply:
I don't agree, if it is trying to create a dependent when its disabled then it should error. if its trying to get the list it should just return empty.

Reply:
Well, I think the service layer should be agnostic of the error handling and logic of how it is used. From that point of view, getting or modifying records just means "don't do anything if disabled". Otherwise, we would need to modify the calling of said functions to depend on the availability of the module.

Reply:
In other words, I think for consistency the service layer should behave the same for retrievals and for updates on disabled modules

Reply:
So my code will say, hey service please insert this, the service will basically ignore my request and say it created it?

Reply:
I agree, that it doesn't make sense. I think the problem is on the responsibility of the service to handle that logic. If the services do error when a call is made on functionality that is not enabled then all calls should error out. That way the developer does not have to figure out what kind of operation is to know if it will error out or not if the module is disabled.